frontend/src/components/EditableLabelsDescriptionListGroup.tsx

```diff
@@ -1,223 +1,191 @@
-import * as React from 'react';
-import {
-  Button,
-  Form,
-  FormGroup,
-  FormHelperText,
-  HelperText,
-  HelperTextItem,
-  Label,
-  LabelGroup,
-  Modal,
-  TextInput,
-} from '@patternfly/react-core';
-import { ExclamationCircleIcon } from '@patternfly/react-icons';
-import DashboardDescriptionListGroup, {
-  DashboardDescriptionListGroupProps,
-} from './DashboardDescriptionListGroup';
+import React, { useState } from 'react';
+import { Label, LabelGroup, Alert, AlertVariant, Button } from '@patternfly/react-core';
+import DashboardDescriptionListGroup from './DashboardDescriptionListGroup';
 
-type EditableTextDescriptionListGroupProps = Partial<
-  Pick<DashboardDescriptionListGroupProps, 'title' | 'contentWhenEmpty'>
-> & {
+interface EditableLabelsProps {
   labels: string[];
-  saveEditedLabels: (labels: string[]) => Promise<unknown>;
-  allExistingKeys?: string[];
+  onLabelsChange: (labels: string[]) => Promise<void>;
   isArchive?: boolean;
-};
+  title?: string;
+  contentWhenEmpty?: string;
+  allExistingKeys: string[];
+}
 
-const EditableLabelsDescriptionListGroup: React.FC<EditableTextDescriptionListGroupProps> = ({
+export const EditableLabelsDescriptionListGroup: React.FC<EditableLabelsProps> = ({
   title = 'Labels',
   contentWhenEmpty = 'No labels',
   labels,
-  saveEditedLabels,
+  onLabelsChange,
   isArchive,
-  allExistingKeys = labels,
+  allExistingKeys,
 }) => {
-  const [isEditing, setIsEditing] = React.useState(false);
-  const [unsavedLabels, setUnsavedLabels] = React.useState(labels);
-  const [isSavingEdits, setIsSavingEdits] = React.useState(false);
+  const [isEditing, setIsEditing] = useState(false);
+  const [isSavingEdits, setIsSavingEdits] = useState(false);
+  const [unsavedLabels, setUnsavedLabels] = useState(labels);
+  const [labelErrors, setLabelErrors] = useState<{ [key: string]: string }>({});
 
-  const editUnsavedLabel = (newText: string, index: number) => {
-    if (isSavingEdits) {
-      return;
+  const validateLabel = (text: string, currentLabel?: string): string | null => {
+    if (currentLabel === text) {
+      return null;
+    }
+
+    const isDuplicate =
+      unsavedLabels.some((key) => key !== currentLabel && key === text) ||
+      allExistingKeys.some((key) => key !== currentLabel && key === text);
+
+    if (isDuplicate) {
+      return `"${text}" already exists. Use a unique name that doesn't match any existing key or property`;
+    }
+    if (text.length > 63) {
+      return "Label text can't exceed 63 characters";
     }
-    const copy = [...unsavedLabels];
-    copy[index] = newText;
-    setUnsavedLabels(copy);
+    return null;
   };
+
+  const handleEditComplete = (
+    _event: MouseEvent | KeyboardEvent,
+    newText: string,
+    currentLabel?: string,
+  ) => {
+    const error = validateLabel(newText, currentLabel);
```

Comment:
@yih-wang Haley, we would need your input in this case on how to handle double duplicates, should we show them both like this? Should we even allow multiple duplicates at edit time? Let me know what you think : this is how it looks with what Mike suggested ( if we want to take this route please let me know which text to use)
![Screenshot 2024-11-18 at 10 59 22 AM](https://github.com/user-attachments/assets/b9c38922-58b9-4ce5-843e-93276bd03843)
![Screenshot 2024-11-18 at 11 05 34 AM](https://github.com/user-attachments/assets/502a2429-aad4-4667-ade0-a827019afa7a)

Reply:
Hi @YuliaKrimerman, I've reviewed this pattern with Anges, who is finalizing the ux validation guidelines. Her suggestion is to allow multiple duplicates at the edit time.

@kaedward, I'd like to hear your input on the validation message here -  Currently, we show the error message below when there's only one label duplicated
`‘mylabel’ already exists. Use a unique name that does not match any existing label or property key.` 

But how should the error message look like when there're multiple duplicates? Should it be a general message like
`Labels duplicated. Use a unique name that does not match any existing label or property key.` 
or a specific message which list all the duplicated labels' names like
`‘mylabel’ and 'new' already exist. Use a unique name that does not match any existing label or property key.`

Reply:
@yih-wang How do these sound? Do we need to specify that the label name must not match a property? 
The label name should be bolded, not in quotation marks. 

ONE ERROR:
**label-name** appears more than once. Ensure that each label is unique.

MORE THAN 1 ERROR:
**label-name** and **label-name2** appear more than once. Ensure that each label is unique.

Reply:
> Do we need to specify that the label name must not match a property?
@kaedward Yes, to be accurate, the label shouldn't match any existing labels or property keys. The pattern you suggested looks good to me. But I'm wondering whether we should say 'label-name already exists' rather than 'label-name appears more than once' because it's already used elsewhere for duplication validation.

Reply:
ONE ERROR:
label-name appears more than once. Ensure that each label is unique and does not match any existing property key.

MORE THAN 1 ERROR:
label-name and label-name2 appear more than once. Ensure that each label is unique and does not match any existing property key.

@yih-wang Here ya go!